pkg/asset/machines/openstack/machines.go

```diff
@@ -140,6 +140,11 @@ func generateProvider(clusterID string, platform *openstack.Platform, mpool *ope
 		})
 	}
 
+	primarySubnet := platform.MachinesSubnet
+	if primarySubnet == "" {
+		primarySubnet = fmt.Sprintf("%s-openshift", clusterID)
```

Comment:
IIRC, `platform.MachinesSubnet` is the ID of the machine subnet, so using a name here would make `PrimarySubnet` attribute inconsistent.
Can we retrieve the subnet ID and pass that to the provider spec?

Reply:
Ok, i just checked. Actually This is all broken, and we cant do it this way :|

Reply:
https://github.com/openshift/cluster-api-provider-openstack/blob/7a86c52029dae191c1aeff96bd18208069172e44/pkg/cloud/openstack/machine/actuator.go#L527